File: app/src/main/java/com/my/televip/virtuals/ui/Components/UItem.java
package com.my.televip.virtuals.ui.Components;

import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;
import com.my.televip.utils.Utils;

import de.robv.android.xposed.XposedHelpers;

public class UItem {

    Object uItem;

    public UItem(Object uItem){
        this.uItem = uItem;
    }

    public int getID(){
        return XposedHelpers.getIntField(uItem, Obfuscate.getFieldName("UItem", "id"));
    }

    public String getText(){
        return Utils.getFieldAsString(XposedHelpers.getObjectField(uItem, Obfuscate.getFieldName("UItem", "text")));
    }

    public String getSubtext(){
        return Utils.getFieldAsString(XposedHelpers.getObjectField(uItem, Obfuscate.getFieldName("UItem", "subtext")));
    }

    public Object getUItem(){
        return uItem;
    }
}

File: app/src/main/java/com/my/televip/virtuals/ui/Components/UItem.kt
package com.my.televip.virtuals.ui.Components

import com.my.televip.obfuscate.Obfuscate
import com.my.televip.utils.Utils
import de.robv.android.xposed.XposedHelpers

class UItem(@JvmField val uItem: Any?) {

    val id: Int
        get() {
            val target = uItem ?: return 0
            return XposedHelpers.getIntField(target, Obfuscate.getFieldName("UItem", "id"))
        }

    val text: String?
        get() {
            val target = uItem ?: return null
            val fieldVal = XposedHelpers.getObjectField(target, Obfuscate.getFieldName("UItem", "text"))
            return Utils.getFieldAsString(fieldVal)
        }

    val subtext: String?
        get() {
            val target = uItem ?: return null
            val fieldVal = XposedHelpers.getObjectField(target, Obfuscate.getFieldName("UItem", "subtext"))
            return Utils.getFieldAsString(fieldVal)
        }
}
